### ISSUE TITLE
Add hat to block definition

### ISSUE THREAD

**Issue by alschmiedt** [google/blockly#3653]
**Is your feature request related to a problem? Please describe.**
The only way to add a hat to a block is either overriding the renderer (adds a hat to all blocks that fit the criteria) or add it to a style. The problem with having it in style is that you have to add two styles with the same colours. 

**Describe the solution you'd like**
Add hat to the block definition.

**Describe alternatives you've considered**

**Additional context**
https://groups.google.com/g/blockly/c/IE4197b9LMY

**Comment by mattbishop** [comment]
StartHat is a feature of all renderers, and it needs a nice, simple way to turn it on and off. Perhaps it could be added with WorkspaceOptions, or better yet, add a new RenderOptions to workspaceOptions to override all the options that ConstantsProvider supplies? I really like Zelos, but there are a few of the sizes I'm going to want to change. I shouldn't have to create a new Renderer class just to tune an existing Renderer.

**Comment by samelhusseini** [comment]
Very much agree with your sentiment @mattbishop, and we should be focusing on ways to make it easier to configure renderers for the next release :)

**Comment by moniika** [comment]
Start Hats can also be added through themes, which is a bit easier.
However, as Sam mentioned, we are actively working on making it easier to configure the rendering constants.

**Comment by rachel-fenichel** [comment]
We resolved this with Sam's PR #3734, which makes it possible to pass renderer overrides in options. For the broader case ("I shouldn't have to create a new Renderer class just to tune an existing Renderer"), we decided that you *do* have to extend the renderer to tune it, and added a [codelab](https://blocklycodelabs.dev/codelabs/custom-renderer/index.html?index=..%2F..index#0) explaining how to do it. There are too many ways to customize a renderer for us to add APIs to satisfy everyone, and we do not want to add them piecewise.
